### DISCUSSION TITLE
Can I use django-polymorphic with django's abstract user model?

### DISCUSSION THREAD

**Discussion by perthvader** [jazzband/django-polymorphic#687] [Q&A]
Currently I have

```
class User(AbstractBaseUser):
    pass

class Foo(User):
    pass

class Bar(User):
    pass
```

Using this snippet https://djangosnippets.org/snippets/1034/, however this causes some trouble with certain orm generated inner joins, see my question here http://stackoverflow.com/questions/25711236/django-orm-fails-to-recognise-concrete-inheritance-in-nested-on-statement

Now I have tried to change my model like this:

```
class User(AbstractBaseUser, PolymorphicModel):
    pass
```

and visa versa but I get _meta errors when migrating, what is the supposed way of achieving the snippets behaviour with django-polymorphic?

**Comment by vdboor** [comment]
I'd expect you can merge both models together, but also make sure to merge the manager and queryset classes. Otherwise, one of the `objects` will get precedence over the other.

> and visa versa but I get _meta errors when migrating

If you can post these errors, it would be really helpful
